cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql

```diff
@@ -41,7 +41,7 @@ where
     // the address escapes into some expression `pointerToLocal`, which flows
     // in a non-trivial way (one or more steps) to a returned expression.
     exists(Expr pointerToLocal |
-      variableAddressEscapesTree(va, pointerToLocal.getFullyConverted()) and
+      variableAddressEscapesTree(va, pointerToLocal) and
```

Comment:
Would it be right to say that the new version is equivalent to:
```
variableAddressEscapesTree(va, pointerToLocal.getFullyConverted()) and
not pointerToLocal.hasConversion()
```
That is, we're not trying to compute something different to before, only exclude those cases where the analysis isn't that great.

Reply:
No, the version in this PR should be better. For example, I'd expect the following code to be caught by the query as of this PR:

```
int myLocal;
void *pointerToLocal = (void *)&myLocal; // has conversion
return pointerToLocal; // data flow step to here
```

For references (rather than pointers), this change will unfortunately exclude all the good results that involve data flow. For example, this good result should be excluded now:

```
int &f() {
  int myLocal;
  int &refToLocal = myLocal; // addr. only escapes to fully-converted expr.
  return refToLocal;
}
```

Reply:
The documentation for `variableAddressEscapesTree` seems to imply that it's second argument must be fully converted.

Reply:
Oh, you're right. That does mean there can't be any conversions. I'll either try to come up with a better fix or make that limitation more clear in the code.